backend/src/Integration/OccupancyStore.hs

```diff
@@ -0,0 +1,33 @@
+module Integration.OccupancyStore where
+
+import Database.Persist (selectList, (==.), PersistStoreWrite (insert))
+import Integration.Init (UserEntityId, EntityField (OccupancyEntityUserId))
+import Database.Persist.Sql (toSqlKey)
+import Database.Persist.Sqlite (runSqlite)
+import Utils.Datatypes (UserId, Occupancy)
+import Utils.Functions (entityToOccupancy, occupancyToEntity)
+
+-- Function to find user occupancies by user ID
+findUserOccupancies :: UserId -> IO (UserId, [Occupancy])
+findUserOccupancies uId = runSqlite "main.db" $ do
+  
+  -- Convert int to UserEntityId
+  let userId = toSqlKey (fromIntegral uId) :: UserEntityId
+  
+  os <- selectList [OccupancyEntityUserId ==. userId] []
+
+  return (uId, map entityToOccupancy os)
```

Comment:
Why return the userid?

Reply:
Since they are the occupancies connected to that Id. I guess it can be removed but might be easier for processing later on.